tests/Functional/FunctionalTest.php

```diff
@@ -66,19 +68,41 @@ public function testFunctional(
             $this->expectExceptionMessage($expectedExceptionMessage);
         }
 
-        $document = $parser->parse($rst);
+        if ($renderMethod === 'renderAll') {
```

Comment:
Both `renderDocument` and `render` are actual methods of the `DocumentNode` class. `renderAll` is not, so this is a bit confusing. Besides, an extra `if` means more complexity, and complexity in tests is something we should avoid (because we don't test the tests).

Maybe instead of a single `tests` directory, we should have  several of them, and then have several tests with several data providers? Something like `tests/render` and `tests/build` (since you are calling `Builder::build()`)?

Reply:
yeah, why not. Just not sure how to implement that

Reply:
I will give it a try.

Reply:
Here is how it could look: https://github.com/linawolf/rst-parser/pull/1